lldbinit.py

```diff
@@ -9,16 +9,174 @@
 import re
 import os
 
-# Bazel builds all binaries from a username-specific cache location
-# for lldb to work properly we need to map that path back to our main source
-project_root = os.path.dirname(os.path.realpath(__file__))
-bazel_workspace = os.path.realpath(os.path.join(project_root, "bazel-sorbet"))
-
-if not(os.path.isdir(bazel_workspace)):
-    print("You need to compile Sorbet first before loading this file")
-else:
-    print("Mapping %s to current working directory" % bazel_workspace)
-    source_map_command = 'settings set -- target.source-map "%s" "%s"' % (bazel_workspace, project_root)
-    ci = lldb.debugger.GetCommandInterpreter()
-    res = lldb.SBCommandReturnObject()
-    ci.HandleCommand(source_map_command, res)
+def __lldb_init_module(debugger, internal_dict):
+    result = lldb.SBCommandReturnObject()
+    init_source_map(debugger, result)
+    ci = debugger.GetCommandInterpreter()
+    ci.HandleCommand("command script add -f lldbinit.cmd_dump_sorbet dumpsorbet", result)
+    register_summary(ci, format_sorbet_core_Name.__name__, "sorbet::core::Name")
+    register_summary(ci, format_sorbet_core_NameRef.__name__, "sorbet::core::NameRef")
+    register_summary(ci, format_sorbet_core_GlobalState.__name__, "sorbet::core::GlobalState")
+    # LLDB will not automatically walk up inheritance chains to find a formatter so it needs to
+    # be specified for every subclass of Instruction
+    register_summary(ci, format_sorbet_object.__name__, "sorbet::cfg::Instruction")
+    register_summary(ci, format_sorbet_object.__name__, "sorbet::cfg::Ident")
+    register_summary(ci, format_sorbet_object.__name__, "sorbet::cfg::Send")
+    register_summary(ci, format_sorbet_object.__name__, "sorbet::cfg::CFG")
+    register_summary(ci, format_sorbet_object.__name__, "sorbet::cfg::VariableUseSite")
+    register_summary(ci, format_sorbet_object.__name__, "sorbet::cfg::LocalRef")
+    register_summary(ci, format_sorbet_object.__name__, "sorbet::core::FileRef")
```

Comment:
Is it intentional that we have entries here for all the `*Ref` classes but `SymbolRef`?

My top three things to print are `TreePtr::loc`, `TypePtr`, and `SymbolRef`.

I think you will find these common eventually too, so I figured I'd suggest adding these while this code is fresh in your mind. I am content to continue to use my normal `p sym.show(gs)` et al commands to print the information I need, so don't feel like you need to also support these three structures if you don't want to.

Reply:
Not intentional, this list was driven by my own debugging so I totally will have skipped completely valid cases (like those you describe). I will add them